src/ctia/task/migration/migrations.clj

```diff
@@ -149,6 +150,91 @@
                (update :incident_time simplify-incident-time))
            doc))))
 
+;; Add new investigation fields
+
+(defn transient-id? [id]
+  (and (string? id)
+       (re-find #"\Atransient" (str id))))
+
+(defn object-id [doc]
+  (let [id (get doc :id)]
+    (when (not (transient-id? id))
+      id)))
+
+(defn read-actions [investigation]
+  (let [actions (get investigation :actions)]
+    (if (string? actions)
+      (json/parse-string actions true)
```

Comment:
I would wrap it in someway just in case it contains garbage values.

Reply:
Got it.